hibernate-reactive-quickstart/src/main/java/org/acme/hibernate/reactive/FruitMutinyResource.java

```diff
@@ -68,39 +72,29 @@
             throw new WebApplicationException("Fruit name was not set on request.", 422);
         }
 
-        // Update function (never returns null)
-        Function<Fruit, Uni<? extends Response>> update = entity -> {
-            entity.setName(fruit.getName());
-            return mutinySession.flush()
-                    .onItem().transform(ignore -> Response.ok(entity).build());
-        };
-
-        return mutinySession
-                .find( Fruit.class, id )
-                      // If entity exists then
-                    .onItem().ifNotNull()
-                        .transformToUni(update)
-                    // else
-                    .onItem().ifNull()
-                        .continueWith(Response.ok().status(404).build());
+        return session
+                .withTransaction(tx -> session.find(Fruit.class, id)
+                    // If entity exists then update it
+                    .onItem().ifNotNull().invoke(entity -> entity.setName(fruit.getName())))
+                .onItem().ifNotNull()
+                    .transform(entity -> Response.ok(entity).build())
+                // If entity not found return the appropriate response
+                .onItem().ifNull()
+                    .continueWith(() -> Response.ok().status(NOT_FOUND).build());
     }
 
     @DELETE
     @Path("{id}")
     public Uni<Response> delete(@RestPath Integer id) {
-        // Delete function (never returns null)
-        Function<Fruit, Uni<? extends Response>> delete = entity -> mutinySession.remove(entity)
-                .chain(mutinySession::flush)
-                .onItem().transform(ignore -> Response.ok().status(204).build());
-
-        return mutinySession
-                .find( Fruit.class, id )
-                    // If entity exists then
+        return session
+                .withTransaction(tx -> session
+                    .find(Fruit.class, id)
+                    // If entity exists then delete it
                     .onItem().ifNotNull()
-                        .transformToUni(delete)
-                    // else
-                    .onItem().ifNull()
-                        .continueWith(Response.ok().status(404).build());
+                        .transformToUni(entity -> session.remove(entity)
+                                .replaceWith(() -> Response.ok().status(NO_CONTENT).build())))
```

Comment:
This is incredibly minor but i would have closed an extra paren on the previous line and avoided the extra nesting/indentation.

Reply:
Since we are discussing this, should I use method reference where possible?
I could rewrite it as:
```
                                .replaceWith(Response.ok().status(NO_CONTENT)::build)
```

Reply:
> This is incredibly minor but i would have closed an extra paren on the previous line and avoided the extra nesting/indentation.

You cannot do that or you would have NO_CONTENT as response instead of NOT_FOUND when the id is not present